src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -113,30 +118,31 @@ public function supportsClass($class)
      */
     protected function loadUser($username, Entry $entry)
     {
-        $password = $this->getPassword($entry);
+        $password = $this->getAttributeValue($entry, $this->passwordAttribute);
 
         return new User($username, $password, $this->defaultRoles);
     }
 
     /**
-     * Fetches the password from an LDAP entry.
+     * Fetches a required unique attribute value from an LDAP entry.
      *
      * @param null|Entry $entry
+     * @param string     $attribute
      */
-    private function getPassword(Entry $entry)
+    private function getAttributeValue(Entry $entry, $attribute)
     {
-        if (null === $this->passwordAttribute) {
+        if (null === $attribute) {
```

Comment:
Actually, I would move this logic inside `loadUser()` instead, as we actually want an exception if we provide `null` as a value for `$attribute`, except if it is a password (if we have no passwordAttribute, this means that we should rely on an authentication mechanism which is not based on password comparison.